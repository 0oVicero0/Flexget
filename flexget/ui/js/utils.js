'use strict';

String.prototype.startsWith = function(prefix) {
  return this.indexOf(prefix) === 0;
};

function registerFlexModule(module) {
  angular.module('flexgetApp').requires.push(module.name);
}

app.menu = [];
app.routes = [];

function register_menu(href, caption, icon, order) {
  href = '/ui/#' + href;
  app.menu.push({href: href, caption: caption, icon: icon, order: order})
}

function register_route(name, url, controller, template) {
  app.routes.push({
    name: name,
    url: url,
    controller: controller,
    template: template
  });
}