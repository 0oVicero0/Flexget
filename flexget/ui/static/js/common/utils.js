'use strict';

String.prototype.startsWith = function(prefix) {
    return this.indexOf(prefix) === 0;
};

function registerFlexModule(module) {
  angular.module('flexgetApp').requires.push(module.name);
}