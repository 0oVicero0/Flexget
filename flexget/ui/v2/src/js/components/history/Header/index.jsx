import React, { Component } from 'react';
import SecondaryNav from 'components/common/SecondaryNav';

class Header extends Component {
  componentDidMount() {

  }

  render() {
    return (
      <SecondaryNav>
        <div />
      </SecondaryNav>
    );
  }
}

export default Header;
