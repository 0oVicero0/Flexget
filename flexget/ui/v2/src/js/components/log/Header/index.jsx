import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import Icon from 'material-ui/Icon';
import Menu, { MenuItem } from 'material-ui/Menu';
import TextField from 'material-ui/TextField';
import Typography from 'material-ui/Typography';

const styleSheet = createStyleSheet('LogHeader', theme => ({
  root: {
    display: 'block',
    [theme.breakpoints.up('sm')]: {
      display: 'flex',
      alignItems: 'baseline',
    },
  },
  grey: {
    color: theme.palette.grey[600],
  },
  spacer: {
    flex: {
      grow: 1,
    },
  },
  textFieldWrapper: {
    display: 'flex',
    alignItems: 'baseline',
  },
  input: {
    marginLeft: 10,
    marginRight: 10,
  },
  menuIcon: {
    paddingRight: 30,
  },
  pointer: {
    cursor: 'pointer',
  },
}));

const ENTER_KEY = 13;

class Header extends Component {
  static propTypes = {
    start: PropTypes.func.isRequired,
    connected: PropTypes.bool.isRequired,
    classes: PropTypes.object.isRequired,
    setQuery: PropTypes.func.isRequired,
    query: PropTypes.string.isRequired,
    clearLogs: PropTypes.func.isRequired,
    lines: PropTypes.string.isRequired,
    setLines: PropTypes.func.isRequired,
  };

  state = {
    open: false,
  };


  componentDidMount() {
    this.start();
  }

  componentWillUnmount() {
    this.stop();
  }

  start = () => { this.stream = this.props.start(); }
  stop = () => this.stream && this.stream.abort()
  reload = () => {
    this.stop();
    this.start();
  }
  clearLogs = () => this.props.clearLogs()
  handleLines = event => this.props.setLines(event.target.value)
  handleQuery = event => this.props.setQuery(event.target.value)
  handleKeyPress = event => event.which === ENTER_KEY && this.reload()
  handleRequestClose = () => this.setState({
    open: false,
    anchorEl: undefined,
  })

  handleMenuClick = event => this.setState({
    open: true,
    anchorEl: event.currentTarget,
  })

  render() {
    const { connected, classes, query, lines } = this.props;
    const { anchorEl, open } = this.state;
    const helperText = 'Supports operators and, or, (), and "str"';

    return (
      <div className={classes.root}>
        <div>
          <Typography type="title">
            Server Log
          </Typography>
          <Typography className={classes.grey} type="subheading">
            { connected ? 'Streaming' : 'Disconnected' }
          </Typography>
        </div>
        <div className={classes.spacer} />
        <div className={classes.textFieldWrapper}>
          <Icon className={`fa fa-filter ${classes.grey}`} />
          <TextField
            id="filter"
            label="Filter"
            className={classes.input}
            value={query}
            onChange={this.handleQuery}
            inputProps={{
              onKeyPress: this.handleKeyPress,
            }}
            helperText={helperText}
          />
          <Icon onClick={this.handleMenuClick} className={`fa fa-ellipsis-v ${classes.grey} ${classes.pointer}`} />
        </div>
        <Menu
          id="log-menu"
          anchorEl={anchorEl}
          open={open}
          onRequestClose={this.handleRequestClose}
        >
          <MenuItem>
            <TextField
              id="lines"
              label="Max Lines"
              value={lines}
              onChange={this.handleLines}
              type="number"
              inputProps={{
                onKeyPress: this.handleKeyPress,
              }}
            />
          </MenuItem>
          <MenuItem onClick={this.clearLogs}>
            <Icon className={`${classes.menuIcon} fa fa-eraser`} />
            Clear
          </MenuItem>
          <MenuItem onClick={connected ? this.stop : this.start}>
            <Icon className={
              classNames(classes.menuIcon, 'fa', {
                'fa-play': !connected,
                'fa-stop': connected,
              })}
            />
            {connected ? 'Stop' : 'Start'}
          </MenuItem>
        </Menu>
      </div>
    );
  }
}

export default withStyles(styleSheet)(Header);

