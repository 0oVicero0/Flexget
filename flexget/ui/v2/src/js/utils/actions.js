export const createAction = namespace => (type, payload, {
  ignore = false,
  message,
} = {}) => ({
  type,
  payload,
  error: (payload instanceof Error),
  meta: {
    ignore,
    message,
    namespace,
  },
});

export const createLoading = namespace => type => ({
  type,
  meta: {
    namespace,
    loading: true,
  },
});
