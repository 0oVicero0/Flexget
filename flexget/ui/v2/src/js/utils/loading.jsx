import Loadable from 'react-loadable';
import Loading from 'components/common/Loader';

export function createAsyncComponent(loader) {
  return Loadable({
    loading: Loading,
    loader,
  });
}

