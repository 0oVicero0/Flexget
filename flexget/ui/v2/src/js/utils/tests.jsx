import React from 'react';
import { MuiThemeProvider } from 'material-ui/styles';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();

export function themed(component) {
  return (
    <MuiThemeProvider>
      { component }
    </MuiThemeProvider>
  );
}

export function router(component) {
  return (
    <MemoryRouter>
      { component }
    </MemoryRouter>
  );
}

export function provider(component, state = {}) {
  return (
    <Provider store={mockStore(state)}>
      { component }
    </Provider>
  );
}
