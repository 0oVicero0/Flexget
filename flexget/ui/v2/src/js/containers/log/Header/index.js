import { connect } from 'react-redux';
import {
  LOG_CONNECT,
  LOG_DISCONNECT,
  LOG_CLEAR,
} from 'actions/log';
import { action, request } from 'utils/actions';
import Header from 'components/log/Header';

export function mapStateToProps({ log }) {
  return {
    connected: log.connected,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    start: payload => dispatch(request(LOG_CONNECT, payload)),
    stop: () => dispatch(request(LOG_DISCONNECT)),
    clearLogs: () => dispatch(action(LOG_CLEAR)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Header);
