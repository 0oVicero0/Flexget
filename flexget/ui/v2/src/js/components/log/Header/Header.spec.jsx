import React from 'react';
import renderer from 'react-test-renderer';
import Header from 'components/log/Header';
import { themed } from 'utils/tests';

describe('components/log/Header', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      themed(<Header
        start={jest.fn()}
        connected
        stop={jest.fn()}
        query=""
        lines="400"
        clearLogs={jest.fn()}
      />)
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
