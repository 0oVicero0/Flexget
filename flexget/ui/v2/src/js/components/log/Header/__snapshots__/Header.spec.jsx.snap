// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/log/Header renders correctly 1`] = `
<div
  className="LogHeader-root-1400901008"
>
  <div>
    <h2
      className="MuiTypography-root-233331062 MuiTypography-title-2106356098"
    >
      Server Log
    </h2>
    <h3
      className="MuiTypography-root-233331062 MuiTypography-subheading-2119063702 LogHeader-grey-968767187"
    >
      Streaming
    </h3>
  </div>
  <div
    className="LogHeader-spacer-3036263360"
  />
  <div
    className="LogHeader-textFieldWrapper-3760854782"
  >
    <span
      aria-hidden="true"
      className="material-icons MuiIcon-root-3402991258 fa fa-filter LogHeader-grey-968767187"
    />
    <div
      className="MuiFormControl-root-3719765767 LogHeader-input-691557382"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
    >
      <label
        className="MuiFormLabel-root-1338640695 MuiInputLabel-root-2962008059 MuiInputLabel-formControl-2201459172 MuiInputLabel-animated-1605689251"
      >
        Filter
      </label>
      <div
        className="MuiInput-root-4262591601 MuiInput-formControl-583691922 MuiInput-inkbar-171024778 MuiInput-underline-892978022"
      >
        <input
          aria-required={undefined}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-3582851417 MuiInput-inputSingleline-2607927053"
          defaultValue={undefined}
          disabled={false}
          id="filter"
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyPress={[Function]}
          onKeyUp={undefined}
          placeholder={undefined}
          rows={undefined}
          type="text"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root-3554229948"
      >
        Supports operators and, or, (), and "str"
      </p>
    </div>
    <span
      aria-hidden="true"
      className="material-icons MuiIcon-root-3402991258 fa fa-ellipsis-v LogHeader-grey-968767187 LogHeader-pointer-501421307"
      onClick={[Function]}
    />
  </div>
</div>
`;
